sycl-fusion/passes/kernel-fusion/SYCLSpecConstMaterializer.h

```diff
@@ -0,0 +1,118 @@
+//==--------------------- SYCLSpecConstMaterializer.h ----------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SYCL_SPEC_CONST_MATERIALIZER_H
+#define SYCL_SPEC_CONST_MATERIALIZER_H
+
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/PassManager.h"
+#include <map>
+#include <set>
+
+namespace llvm {
+class Function;
+
+///
+/// Utility pass to insert specialization constants values into the module as a
+/// metadata node.
+class SYCLSpecConstDataInserter
+    : public PassInfoMixin<SYCLSpecConstDataInserter> {
+public:
+  SYCLSpecConstDataInserter(const unsigned char *SpecConstData,
+                            size_t SpecConstDataSize)
+      : SpecConstData(SpecConstData), SpecConstDataSize(SpecConstDataSize) {};
+
+  SYCLSpecConstDataInserter() {
+    SpecConstDataSize = 255;
+    for (unsigned i = 0; i < SpecConstDataSize; ++i)
+      DebugSpecConstData.push_back(i);
+
+    SpecConstData = DebugSpecConstData.data();
+  }
```

Comment:
```suggestion
  SYCLSpecConstDataInserter() : DebugSpecConstData(255), SpecConstData(DebugSpecConstData) {
    std::iota(DebugSpecConstData.begin(), DebugSpecConstData.end(), 0);
  }
```

Reply:
We actually are not using this at all, removed the `DebugSpecConstData` altogether.